src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -1228,6 +1238,7 @@ public static void initClientHooks(Minecraft mc, ReloadableResourceManager resou
         initializedClientHooks = true;
 
         ForgeGameTestHooks.registerGametests();
+        ModLoader.get().postEvent(new RegisterSpriteSourceTypesEvent(SPRITE_SOURCE_TYPES_MAP));
```

Comment:
With the way the event is currently fired here, it doesn't run in datagen because datagen doesn't instantiate `Minecraft`. Custom `SpriteSource`s being registered is required though if you want to be able to datagen sources with `SpriteSourceProvider`